### DISCUSSION TITLE
iOS error: 'WebRTC/WebRTC.h' file not found

### DISCUSSION THREAD

**Discussion by ycherniavskyi** [flutter-webrtc/flutter-webrtc#1026] [Q&A] [answered]
Just want to summarise all solutions of this error (and close relevant issues #755, #966, #773, #713, #892).

**Comment by ycherniavskyi** [comment, marked as answer]
### Reason

So the reason of this error is actually this warning:
```
warning: [CP] WebRTC.xcframework: Unable to find matching slice in 'ios-arm64_armv7 ios-arm64_x86_64-simulator' for the current build architectures (arm64 x86_64 i386) and platform (-iphonesimulator).
```

### Solution

Add in file `./ios/Podfile` next snippet:
```ruby
#...
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    # snippet begin
    if target.name == "WebRTC-SDK"
      target.build_configurations.each do |build_configuration|
        build_configuration.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = '$(inherited) i386'
      end
    end
    # snippet end
  end
end
```

### Note

For sure, this error could be resolved in different ways. Provided in solution way is just analog what Flutter does for [Flutter plugins](https://github.com/flutter/flutter/blob/5d31b07ed5f8ef4a5a3be5cef01138fd0d59e062/packages/flutter_tools/bin/podhelper.rb#L83). But because WebRTC-SDK technically is [not a Flutter plugin](https://github.com/flutter/flutter/blob/5d31b07ed5f8ef4a5a3be5cef01138fd0d59e062/packages/flutter_tools/bin/podhelper.rb#L58), then this must be done manually.

If someone would like to explain why XCode wants to build i386 architecture, then welcome in the comments 😉.

**Reply by tinhnvc-gadget** [reply]
> ### Reason
> So the reason of this error is actually this warning:
> 
> ```
> warning: [CP] WebRTC.xcframework: Unable to find matching slice in 'ios-arm64_armv7 ios-arm64_x86_64-simulator' for the current build architectures (arm64 x86_64 i386) and platform (-iphonesimulator).
> ```
> 
> ### Solution
> Add in file `./ios/Podfile` next snippet:
> 
> ```ruby
> #...
> post_install do |installer|
>   installer.pods_project.targets.each do |target|
>     flutter_additional_ios_build_settings(target)
>     # snippet begin
>     if target.name == "WebRTC-SDK"
>       target.build_configurations.each do |build_configuration|
>         build_configuration.build_settings['EXCLUDED_ARCHS[sdk=iphonesimulator*]'] = '$(inherited) i386'
>       end
>     end
>     # snippet end
>   end
> end
> ```
> 
> ### Note
> For sure, this error could be resolved in different ways. Provided in solution way is just analog what Flutter does for [Flutter plugins](https://github.com/flutter/flutter/blob/5d31b07ed5f8ef4a5a3be5cef01138fd0d59e062/packages/flutter_tools/bin/podhelper.rb#L83). But because WebRTC-SDK technically is [not a Flutter plugin](https://github.com/flutter/flutter/blob/5d31b07ed5f8ef4a5a3be5cef01138fd0d59e062/packages/flutter_tools/bin/podhelper.rb#L58), then this must be done manually.
> 
> If someone would like to explain why XCode wants to build i386 architecture, then welcome in the comments 😉.

I have the same error with flutter_webrtc: 0.9.12. 
Please check again, thanks

**Comment by mwk24** [comment]
This did not resolve my issue, which was building using "flutter build ipa". 

I fixed it thusly.... (seems webRTC doesn't support arm7, so wouldn't be possible to build for that)

<img width="655" alt="image" src="https://user-images.githubusercontent.com/30983/186180754-2a641a45-c196-4136-be78-baf4792a5662.png">

**Reply by ycherniavskyi** [reply]
I could say even more - Flutter drop support of [32-bit iOS devices](https://docs.google.com/document/d/1cc5EOsuTlbf4dTDTwmkD3aKjS8XEbVCIqi9BFct9XHM/edit?resourcekey=0-Iv0gXDx7nSDCe3YDfxDKqw).

**Comment by lyskouski** [comment]
Raised back with Flutter 3.13.6 for "flutter build ipa".
